python/ray/air/checkpoint.py

```diff
@@ -560,7 +636,20 @@ def from_uri(cls, uri: str) -> "Checkpoint":
         Returns:
             Checkpoint: checkpoint object.
         """
-        return cls(uri=uri)
+        state = {}
+        try:
+            checkpoint_metadata_uri = os.path.join(uri, _CHECKPOINT_METADATA_FILE_NAME)
+            metadata = pickle.loads(read_file_from_uri(checkpoint_metadata_uri))
+            cls._check_type_compatability(metadata.checkpoint_type)
+            cls = metadata.checkpoint_type
+            state = metadata.checkpoint_state
+        except FileNotFoundError:
+            pass
+
+        checkpoint = cls(uri=uri)
+        checkpoint.__dict__.update(state)
```

Comment:
Can we consolidate the logic from `from_bytes`, `from_uri`, and `from_directory` to a common function?

Reply:
Do you mean this logic?

```
checkpoint = cls(uri=uri)
checkpoint.__dict__.update(state)
```

`from_bytes` doesn't explicitly load metadata.

`from_uri`, `from_directory`, and `from_dict` load metadata differently, so we can't define a single function for that.

Reply:
Got it ok it's just 2 lines of common code, so ok to leave as is!